bugsnag-plugin-android-anr/src/main/java/com/bugsnag/android/AnrPlugin.kt

```diff
@@ -9,31 +9,53 @@ internal class AnrPlugin : Plugin {
         private const val LOAD_ERR_MSG = "Native library could not be linked. Bugsnag will " +
             "not report ANRs. See https://docs.bugsnag.com/platforms/android/anr-link-errors"
 
-        /**
-         * Returns the index of the JVM frame that led to the native ANR, or -1 if not found.
-         */
-        internal fun getNativeANRIndex(stackTrace: Array<StackTraceElement>): Int {
-            // This is not perfect, because some JVM calls are technically native (like sleep).
-            val notifyIndex = stackTrace.indexOfFirst { frame ->
-                frame.methodName == "notifyAnrDetected" && frame.className == "com.bugsnag.android.AnrPlugin"
-            }
-            if (notifyIndex < 0) {
-                return -1
+        internal fun endsInNativeCall(stackTrace: Array<StackTraceElement>): Boolean {
+            if (stackTrace.isEmpty()) {
+                return false
             }
+            return stackTrace.first().isNativeMethod
+        }
 
-            val nativeAnrIndex = notifyIndex + 1
-            if (nativeAnrIndex >= stackTrace.size || !stackTrace[nativeAnrIndex].isNativeMethod) {
-                return -1
+        internal fun hasReservedNativeFrames(stackTrace: List<NativeStackframe>): Boolean {
+            val index = stackTrace.indexOfFirst { frame ->
+                val method = frame.method
+                method != null && method.startsWith("_Z")
```

Comment:
May be worth extracting "_Z" into a constant variable and adding a quick comment about its significance - I initially missed the longer comment below and got confused about what this was.

You can create a constant in Kotlin using either of the following approaches: https://stackoverflow.com/a/46729715